src/main/java/me/champeau/gradle/japicmp/ignore/entity/ClassDescriptor.java

```diff
@@ -0,0 +1,135 @@
+package me.champeau.gradle.japicmp.ignore.entity;
+
+import japicmp.model.JApiClass;
+import japicmp.model.JApiConstructor;
+import japicmp.model.JApiField;
+import japicmp.model.JApiMethod;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClassDescriptor {
+  private final String fullName;
+
+  private final Map<String, RemoveMethodEntity> removedMethods = new HashMap<>();
+  private final Map<String, ChangeMethodEntity> changedMethods = new HashMap<>();
+
+  private final Map<String, RemoveFieldEntity> removedField = new HashMap<>();
+  private final Map<String, ChangeFieldEntity> changedFields = new HashMap<>();
+
+  private final Map<String, RemoveConstructorEntity> removedConstructors = new HashMap<>();
+  private final Map<String, ChangeConstructorEntity> changedConstructors = new HashMap<>();
+
+  private final Map<String, RemoveClassEntity> removeClasses = new HashMap<>();
+
+
+  public ClassDescriptor(String packageClassName) {
+    this.fullName = packageClassName;
+  }
+
+  private void fill(List<Entity> entities) {
+    for (Entity entity : entities) {
+      add(entity);
+    }
+  }
+
+  public void add(Entity entity) {
+    if (entity instanceof RemoveMethodEntity) {
+      RemoveMethodEntity castedEntity = (RemoveMethodEntity) entity;
+      removedMethods.put((castedEntity).getName(), castedEntity);
+    } else if (entity instanceof ChangeMethodEntity) {
+      ChangeMethodEntity castedEntity = (ChangeMethodEntity) entity;
+      changedMethods.put(castedEntity.getName(), castedEntity);
+    } else if (entity instanceof RemoveFieldEntity) {
+      RemoveFieldEntity castedEntity = (RemoveFieldEntity) entity;
+      removedField.put((castedEntity).getName(), castedEntity);
+    } else if (entity instanceof ChangeFieldEntity) {
+      ChangeFieldEntity castedEntity = (ChangeFieldEntity) entity;
+      changedFields.put((castedEntity).getFirstFieldName(), castedEntity);
+      changedFields.put((castedEntity).getSecondFieldName(), castedEntity);
+    } else if (entity instanceof RemoveConstructorEntity) {
+      RemoveConstructorEntity castedEntity = (RemoveConstructorEntity) entity;
+      removedConstructors.put((castedEntity).getName(), castedEntity);
+    } else if (entity instanceof ChangeConstructorEntity) {
+      ChangeConstructorEntity castedEntity = (ChangeConstructorEntity) entity;
+      changedConstructors.put((castedEntity).getName(), castedEntity);
+    } else if (entity instanceof RemoveClassEntity) {
+      RemoveClassEntity castedEntity = (RemoveClassEntity) entity;
+      removeClasses.put(castedEntity.getClassName(), castedEntity);
+    }
```

Comment:
Any way to avoid `instanceof`?

Reply:
I would like to leave this as a technical debt for now. Didn't find easy way to avoid it.

Reply:
Well, how about a Visitor?